src/textual/widgets/_data_table.py

```diff
@@ -293,18 +304,26 @@ def watch_cursor_cell(
     def _highlight_cell(self, coordinate: Coordinate) -> None:
         """Apply highlighting to the cell at the coordinate, and emit event."""
         self.refresh_cell(*coordinate)
-        cell_value = self.get_cell_value(coordinate)
-        self.emit_no_wait(DataTable.CellHighlighted(self, cell_value, coordinate))
+        try:
+            cell_value = self.get_cell_value(coordinate)
+        except CellDoesntExist:
+            # The cell may not exist e.g. when the table is cleared.
+            # In that case, there's nothing for us to do here.
+            return
+        else:
+            self.emit_no_wait(DataTable.CellHighlighted(self, cell_value, coordinate))
 
     def _highlight_row(self, row_index: int) -> None:
         """Apply highlighting to the row at the given index, and emit event."""
         self.refresh_row(row_index)
-        self.emit_no_wait(DataTable.RowHighlighted(self, row_index))
+        if row_index in self.data:
+            self.emit_no_wait(DataTable.RowHighlighted(self, row_index))
```

Comment:
`self.refresh_row` will also check that we are trying to refresh a row that exists, so is that why you don't have the call to `refresh_row` inside the if statement?

Reply:
Yep, that's correct.